command/agent/http.go

```diff
@@ -511,6 +511,9 @@ func (s *HTTPServer) wrap(handler func(resp http.ResponseWriter, req *http.Reque
 				} else if strings.HasSuffix(errMsg, structs.ErrJobRegistrationDisabled.Error()) {
 					errMsg = structs.ErrJobRegistrationDisabled.Error()
 					code = 403
+				} else if strings.HasSuffix(errMsg, structs.ErrIncompatibleFiltering.Error()) {
+					errMsg = structs.ErrIncompatibleFiltering.Error()
+					code = 400
```

Comment:
I don't know if this is the right place to handle this, but it would be good to have it a code path that is common to all endpoints.

Reply:
It is absolutely a weird place to jam it and not the API (HasSuffix?!) any of us would prefer but... it's idiomatic for now. :+1: